src/NUnitFramework/framework/Constraints/CollectionEquivalentConstraintResult.cs

```diff
@@ -46,16 +46,13 @@ public class CollectionEquivalentConstraintResult : ConstraintResult
             bool isSuccess)
             : base(constraint, actual, isSuccess)
         {
-            if (tallyResult == null)
-            {
-                throw new System.ArgumentNullException("Tally was null.");
-            }
+            Guard.ArgumentNotNull(tallyResult, "tallyResult");
 
             _tallyResult = tallyResult;
         }
 
         /// <summary>Write the custom failure message for this object's <see cref="CollectionEquivalentConstraint"/>.</summary>
-        /// <param name="writer"></param>
+        /// <param name="writer">The object to write the failure message to.</param>
```

Comment:
Not sure why I wrote object instead of the actual type I'm using to write the msg to... sorry about that!